### ISSUE TITLE
"Command is not "create", executing runc doing nothing"

### ISSUE THREAD

**Issue by Mercy811** [NVIDIA/nvidia-container-runtime#83]
i ran a `nmtwizard/opennmt-tf` container. But i only ran for a few seconds and stop automatically. 

- i checked `/var/log/nvidia-container-runtime.log`. **It seemed that there was something wrong with my runtime since it was looking for 'runc' and 'docker-runc' repeatly**

- according to [this](https://github.com/docker/for-linux/issues/486) The docker- prefix has been removed from these binaries since docker 18.09.0. 
- i've registered `nvidia` runtime by daemon configuration file

files to attache
---
**/var/log/nvidia-container-runtime.log**
```
2019/10/14 16:29:58 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:29:58 Using bundle file: /run/containerd/io.containerd.runtime.v1.linux/moby/47e9f2194316d67a60bfb9ec9eebbdcdf723c66b41f34945125eaed9c08e9ddb/config.json
2019/10/14 16:29:58 prestart hook path: /usr/bin/nvidia-container-runtime-hook
2019/10/14 16:29:58 existing nvidia prestart hook in OCI spec file
2019/10/14 16:29:58 Prestart hook added, executing runc
2019/10/14 16:29:58 Looking for "docker-runc" binary
2019/10/14 16:29:58 "docker-runc" binary not found
2019/10/14 16:29:58 Looking for "runc" binary
2019/10/14 16:29:58 Runc path: /usr/bin/runc
2019/10/14 16:29:59 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:29:59 Command is not "create", executing runc doing nothing
2019/10/14 16:29:59 Looking for "docker-runc" binary
2019/10/14 16:29:59 "docker-runc" binary not found
2019/10/14 16:29:59 Looking for "runc" binary
2019/10/14 16:29:59 Runc path: /usr/bin/runc
2019/10/14 16:29:59 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:29:59 Command is not "create", executing runc doing nothing
2019/10/14 16:29:59 Looking for "docker-runc" binary
2019/10/14 16:29:59 "docker-runc" binary not found
2019/10/14 16:29:59 Looking for "runc" binary
2019/10/14 16:29:59 Runc path: /usr/bin/runc
2019/10/14 16:29:59 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:29:59 Command is not "create", executing runc doing nothing
2019/10/14 16:29:59 Looking for "docker-runc" binary
2019/10/14 16:29:59 "docker-runc" binary not found
2019/10/14 16:29:59 Looking for "runc" binary
2019/10/14 16:29:59 Runc path: /usr/bin/runc
2019/10/14 16:29:59 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:29:59 Command is not "create", executing runc doing nothing
2019/10/14 16:29:59 Looking for "docker-runc" binary
2019/10/14 16:29:59 "docker-runc" binary not found
2019/10/14 16:29:59 Looking for "runc" binary
2019/10/14 16:29:59 Runc path: /usr/bin/runc
2019/10/14 16:30:01 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:30:01 Command is not "create", executing runc doing nothing
2019/10/14 16:30:01 Looking for "docker-runc" binary
2019/10/14 16:30:01 "docker-runc" binary not found
2019/10/14 16:30:01 Looking for "runc" binary
2019/10/14 16:30:01 Runc path: /usr/bin/runc
2019/10/14 16:30:01 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:30:01 Command is not "create", executing runc doing nothing
2019/10/14 16:30:01 Looking for "docker-runc" binary
2019/10/14 16:30:01 "docker-runc" binary not found
2019/10/14 16:30:01 Looking for "runc" binary
2019/10/14 16:30:01 Runc path: /usr/bin/runc
2019/10/14 16:30:01 Running /usr/bin/nvidia-container-runtime
2019/10/14 16:30:01 Command is not "create", executing runc doing nothing
2019/10/14 16:30:01 Looking for "docker-runc" binary
2019/10/14 16:30:01 "docker-runc" binary not found
2019/10/14 16:30:01 Looking for "runc" binary
2019/10/14 16:30:01 Runc path: /usr/bin/runc
```

**/etc/docker/daemon.json**
```
{
    "runtimes": {
        "nvidia": {
            "path": "/usr/bin/nvidia-container-runtime",
            "runtimeArgs": []
        }
    }
}
```

**Comment by andrewssobral** [comment]
same for me

**Comment by kelly8080** [comment]
Here the same problem.
Try to add 
`deafult-runtime: "nvidia"`
inside  daemon.json

**Comment by elezar** [comment]
These are not errors, but indicating the `nvidia-container-runtime` is searching the path for the specified `runc` executables. @andrewssobral in the `podman` case you should ensure that one of these (i.e. `runc`) is in the path. We are looking at making this more configurable through a config option in the near future.

@Mercy811 with regards to the `command is not "create"` message that you're seeing on the logs. All `runc` (sub-)commands are forwarded by `docker` (or `podman`) to the `nvidia-container-runtime` but only the `create` subcommand requires that the OCI spec be modified. We will look at whether we can improve the logging around this to make it less verbose.

**Comment by PriamX** [comment]
@elezar Thank you for the explanation. I had assumed that "command is not create" was some kind of error. It's good to know it's working as designed. The verbose logging is a minor hassle, and no hassle after setting up the logrotate, but I do appreciate you're looking at improving the logging.

**Comment by elezar** [comment]
We have just released `v1.10.0-rc.2` of the NVIDIA Container Toolkit to our [experimental repositories](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit). This offers some enhancements relevant to this issue:
* It reduces log verbosity by switching to debug logging for most output
* It fixes an issue we had when locating the low-level runtimes if the PATH is was not set. We saw this behaviour when specifying the `nvidia-container-runtime` as a runtime in `podman`
* It adds support for specifying a low-level runtime (or runtime candidates) directly in the `nvidia-container-runtime.runtimes` config option.

@Mercy811 could you check whether this addresses your original issue.
